src/Tools/dotnet-monitor/CollectionRules/CollectionRuleEndpointInfoSourceCallbacks.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules
+{
+    internal class CollectionRuleEndpointInfoSourceCallbacks :
+        IEndpointInfoSourceCallbacks
+    {
+        private readonly CollectionRuleService _service;
+
+        public CollectionRuleEndpointInfoSourceCallbacks(CollectionRuleService service)
+        {
+            _service = service ?? throw new ArgumentNullException(nameof(service));
+        }
+
+        public Task OnAddedEndpointInfoAsync(IEndpointInfo endpointInfo, CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
+        }
+
+        public Task OnBeforeResumeAsync(IEndpointInfo endpointInfo, CancellationToken cancellationToken)
+        {
+            return _service.ApplyRules(endpointInfo, cancellationToken);
```

Comment:
will this still be called on nosuspend ports?

Reply:
Yes. There is no way for us to tell if a port is suspended or not. We have to accept the connection, do whatever setup is necessary, and always call ResumeRuntime.